javascript/extractor/src/com/semmle/js/extractor/CFGExtractor.java

```diff
@@ -774,13 +774,13 @@ public Finally(SourceLocation loc, BlockStatement body) {
 		}
 
 		// associate statements with their (direct or indirect) labels
-		private final Map<Statement, Set<String>> loopLabels = new LinkedHashMap<Statement, Set<String>>();
+		private Map<Statement, Set<String>> loopLabels = new LinkedHashMap<Statement, Set<String>>();
```

Comment:
Perhaps document that these three caches are replaced and restored at each function.

Reply:
Good plan.